### PR TITLE
remove as XXXX of TypeScript (file: fix-203-banned_word.ts)

### PR DESCRIPTION
**Pull request by vibh1103**
Fix #203

### CODE REVIEW

File: apps/page/option/banned_word.ts [hunk 1]
```diff
@@ -28,25 +28,25 @@ export function toBannedTarget(value: string): BannedTarget {
...
-    ) as HTMLInputElement;
-    this.addText = document.getElementById(
-      "bannedWordAddText",
-    ) as HTMLInputElement;
-    this.wordList = document.getElementById("bannedWordList") as HTMLDivElement;
-
-    this.addButton.addEventListener("click", async () => {
+    this.addButton = document.getElementById("bannedWordAddButton");
+    this.addText = document.getElementById("bannedWordAddText");
+    this.wordList = document.getElementById("bannedWordList");
```
Comment (ikemo3) [inline apps/page/option/banned_word.ts:41]: Please check for the existence of the element at this location.

File: apps/page/option/banned_word.ts [hunk 2]
```diff
@@ -28,25 +28,25 @@ export function toBannedTarget(value: string): BannedTarget {
...
+
+    this.addButton?.addEventListener("click", async () => {
+      if (!(this.addText instanceof HTMLInputElement)) {
+        throw new Error("bannedWordAddText is not HTMLInputElement");
+      }
       const word = this.addText.value;
       if (word === "") {
-        return;
+        throw new Error("bannedWordAddText is not HTMLInputElement");
```
Comment (ikemo3) [inline apps/page/option/banned_word.ts:49]: The "word" here refers to a string input by the user, so it's simply being ignored.
Please revert the change.

File: apps/page/option/banned_word.ts [hunk 3]
```diff
@@ -75,15 +75,23 @@ export default class BannedWords {
   }
 
   public clear(): void {
-    this.wordList.innerHTML = "";
+    if (this.wordList instanceof HTMLDivElement) {
+      this.wordList.innerHTML = "";
+    } else {
+      Logger.debug("bannedWordList is not HTMLDivElement");
+    }
```
Comment (ikemo3) [inline apps/page/option/banned_word.ts:82]: If you check in the constructor, the existence check for this element will become unnecessary.

File: apps/page/option/banned_word.ts [hunk 4]
```diff
@@ -75,15 +75,23 @@ export default class BannedWords {
...
   }
 
   public async load(): Promise<void> {
-    const words: BannedWord[] = await BannedWordRepository.load();
-    this.wordList.innerHTML = "";
+    if (this.wordList instanceof HTMLDivElement) {
```
Comment (ikemo3) [inline apps/page/option/banned_word.ts:86]: Here and all below are the same as above.

**Review by ikemo3 — CHANGES_REQUESTED** [review]

File: apps/page/option/banned_word.ts [hunk 5]
```diff
@@ -64,6 +66,36 @@ export default class BannedWords {
     });
   }
 
+  private assertButtonElement(element: HTMLElement | null): HTMLButtonElement {
```
Comment (ikemo3) [inline apps/page/option/banned_word.ts:69]: This approach is overly verbose. Whether it's null or defined as a different element, it's a programming error and the same error should suffice. Just adding a check like `if (!(anchor instanceof HTMLAnchorElement)) {` in the constructor, as written in #931, should be enough. Please don't define a function just for that.

**Review by ikemo3 — CHANGES_REQUESTED** [review]

File: apps/page/option/banned_word.ts (hunk 5 above)
Comment (vibh1103) [reply apps/page/option/banned_word.ts:69]: In this case we are getting element from `getElementById` method is having type of HTMLElement and anyhow we need to do typecasting using as to convert it into `HTMLButtonElement` or required element. So i think we need to revert the changes just add changes like this to avoid error`if (!(button instanceof HTMLButtonElement)) {`.

File: apps/page/option/banned_word.ts [hunk 6]
```diff
@@ -64,6 +66,36 @@ export default class BannedWords {
...
+    if (!element) {
+      throw new Error("bannedWordAddButton is null");
+    }
+    if (element instanceof HTMLButtonElement) {
+      return element as HTMLButtonElement;
```
Comment (ikemo3) [inline apps/page/option/banned_word.ts:74]: `as HTMLButtonElement` is redundant.
The same applies to the other two locations.

File: apps/page/option/banned_word.ts (hunk 5 above)
Comment (ikemo3) [reply apps/page/option/banned_word.ts:69]: OK, I understand. 
To put it into an instance variable, it needs to be non-null at this point.
Your approach is correct, and the function can also be reused commonly.
I found a redundant "as" casting. Please fix just that.

**Review by ikemo3 — APPROVED** [review]
LGTM!
